coderd/httpmw/organizationparam.go

```diff
@@ -53,26 +53,40 @@ func ExtractOrganizationParam(db database.Store) func(http.Handler) http.Handler
 			}
 
 			var organization database.Organization
-			var err error
-			// Try by name or uuid.
-			id, err := uuid.Parse(arg)
-			if err == nil {
-				organization, err = db.GetOrganizationByID(ctx, id)
+			var dbErr error
+
+			// If the name is exactly "default", then we fetch the default
+			// organization. This is a special case to make it easier
+			// for single org deployments.
+			//
+			// arg == uuid.Nil.String() should be a temporary workaround for
+			// legacy provisioners that don't provide an organization ID.
+			// This prevents a breaking change.
+			// TODO: This change was added March 2024, this should be removed
+			//		some number of months after that date.
+			if arg == codersdk.DefaultOrganization || arg == uuid.Nil.String() {
+				organization, dbErr = db.GetDefaultOrganization(ctx)
```

Comment:
This is the only quirky thing. A nil uuid has to be interpreted as the default org for backwards compatibility.

I've added a `default` string so we can use that placeholder in single org deployments and not have to use a uuid which is harder to remember.